src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -289,34 +289,38 @@ impl CpuInfo {
     pub fn adjust_config(&mut self) -> Result<()> {
         let features: Vec<&str> = self.cpu_features.split(',').map(|v| v.trim()).collect();
         self.cpu_features = features.join(",");
-        Ok(())
-    }
-
-    /// Validate the configuration information.
-    pub fn validate(&self) -> Result<()> {
-        Ok(())
-    }
 
-    /// Get default number of guest vCPUs.
-    pub fn get_default_vcpus(&self) -> u32 {
         let cpus = num_cpus::get() as u32;
+
+        // adjust default_maxvcpus
+        if self.default_maxvcpus == 0 || self.default_maxvcpus > cpus {
+            self.default_maxvcpus = cpus;
+        }
+
+        // adjust default_vcpus
         if self.default_vcpus < 0 || self.default_vcpus as u32 > cpus {
-            cpus
+            self.default_vcpus = cpus as i32;
         } else if self.default_vcpus == 0 {
-            default::DEFAULT_GUEST_VCPUS
-        } else {
-            self.default_vcpus as u32
+            self.default_vcpus = default::DEFAULT_GUEST_VCPUS as i32;
         }
+
+        if self.default_vcpus > self.default_maxvcpus as i32 {
+            self.default_vcpus = self.default_maxvcpus as i32;
+        }
+
+        Ok(())
     }
 
-    /// Get default maximal number of guest vCPUs.
-    pub fn get_default_max_vcpus(&self) -> u32 {
-        let cpus = num_cpus::get() as u32;
-        if self.default_maxvcpus == 0 || self.default_maxvcpus > cpus {
-            cpus
-        } else {
-            self.default_maxvcpus
+    /// Validate the configuration information.
+    pub fn validate(&self) -> Result<()> {
+        if self.default_vcpus > self.default_maxvcpus as i32 {
```

Comment:
What if `self.default_vcpus == 0` or `self.default_maxvcpus ==0`.

Reply:
If the call sequence is correct, there will be no such case, neither of them will be zero. Even the validate() should be reverted, because `default_maxvcpus` is ensured to be greater or equal to `default_vcpus`.